mrp_unbuild_bom_cust_qty/models/mrp_unbuild_bom_quants.py

```diff
@@ -0,0 +1,99 @@
+# © 2021 Solvos Consultoría Informática (<http://www.solvos.es>)
+# License LGPL-3.0 (http://www.gnu.org/licenses/lgpl-3.0.html)
+
+from odoo import models, fields, api, _
+from odoo.exceptions import ValidationError
+
+
+class MrpUnbuildBoMQuants(models.Model):
+    _name = "mrp.unbuild.bom.quants"
+    _inherit = "mrp.unbuild.bom.mixin"
+    _description = "Quants for BoM based Unbuild"
+
+    custom_qty = fields.Float(
+        string="Quantity",
+        digits='Product Unit of Measure',
+        required=True,
+    )
+    lot_id = fields.Many2one(
+        comodel_name="stock.lot",
+    )
+    has_lot = fields.Boolean(
+        compute="_compute_has_lot",
+    )
+    lot_required = fields.Boolean(
+        compute="_compute_lot_required",
+    )
+    departure_date = fields.Datetime(
+        readonly=True
+    )
+
+    @api.depends("lot_id")
+    def _compute_has_lot(self):
+        for record in self:
+            record.has_lot = bool(record.lot_id)
+
+    @api.depends("bom_line_id")
+    def _compute_lot_required(self):
+        for record in self:
+            record.lot_required = (
+                record.bom_line_id.product_id.tracking != "none"
+            )
+
+    @api.constrains("custom_qty")
+    def _check_custom_qty(self):
+        for record in self:
+            if record.custom_qty <= 0.0:
+                raise ValidationError(
+                    _("Invalid quant for %s: it must be greater than zero!") %
+                    record.name
+                )
```

Comment:
Could we change to this?
```suggestion
                    _("Invalid quant for %(name)s: it must be greater than zero!",
                    name=record.name)
                )
```

Reply:
Not at this time...